src/pybind/mgr/volumes/fs/async_job.py

```diff
@@ -49,7 +49,7 @@ def run(self):
                             for m in missing:
                                 self.async_job.jobs[m] = []
                                 self.async_job.q.append(m)
-                        vol_job = self.async_job.get_job()
+                        vol_job = self.async_job.get_job() if self.vc.mgr.has_fs_map else None
```

Comment:
This just narrows the race window. IMO, we should check if its possible to pass a custom mount timeout (something like 10 seconds) for mounts required by mgr/volumes, since its safe to timeout the mount process since it gets retried. That would prevent the finisher thread in ceph-mgr to be blocked for long time.